rust/processor/src/bq_analytics/generic_parquet_processor.rs

```diff
@@ -153,6 +159,8 @@ where
         if self.buffer_size_bytes >= max_buffer_size {
             let start_version = self.buffer.first().unwrap().version();
             let end_version = self.buffer.last().unwrap().version();
```

Comment:
We know for sure that the buffer has at least one item in it?

Reply:
it should, and I have a follow-up pr that prevents this happening too